### ISSUE TITLE
istanbul: 1.0.0-alpha.2 : File [/foo/bar.js] ignored, nothing could be mapped

### ISSUE THREAD

**Issue by yvele** [gotwarlost/istanbul#572]
Running **istanbul 1.0.0-alpha.2** on babel generated code with [source-maps] option set to (https://babeljs.io/docs/usage/cli/) [both](https://babeljs.io/docs/usage/options/) (babel 6.6.5, babel-core 6.7.2) I got many warnings like that:

```
File [/Users/Yves/foo/bar.js] ignored, nothing could be mapped
```

Here is how I run istanbul:

```
istanbul cover _mocha -- ./test --compilers js:babel-register \
   --require babel-polyfill \
   --require should \
   --require sinon
```

**Comment by codekirei** [comment]
Just ran into this as well. I believe it has to do with named vs anonymous `default` exports.

I put together a super simple example repo that reproduces this [here](https://github.com/codekirei/istanbul-export-default-coverage).

- [named](https://github.com/codekirei/istanbul-export-default-coverage/blob/master/src/covered.js) &mdash; gets covered
- [anonymous](https://github.com/codekirei/istanbul-export-default-coverage/blob/master/src/not-covered.js) &mdash; throws `File ... ignored` warning and is not covered

My current workaround is to just structure my code with named exports.

**Comment by gonsfx** [comment]
Same here. Thanks for the report!

**Comment by laggingreflex** [comment]
I'm getting errors in files whose only contents are named export statements, but of the following form:
```
/* eslint-disable */

// @create-index

export handle from './handle.js';
export throwEx from './throwEx.js';
```
(created via [create-index](https://github.com/gajus/create-index))

**Comment by chalda** [comment]
any resolution to this? I don't want to create a massive commit changing every index.js export in my project...

**Comment by yvele** [comment]
I'm switching to https://github.com/facebook/jest

**Comment by bengrunfeld** [comment]
@yvele after x hours (too embarrassed to say exactly how long) of trying to set up `mocha` and `nyc` to run the simplest possible unit test and generate a coverage report with ES6 `import` and `export default`, I tried going with your suggestion, and within 10 minutes, everything just freaking worked.

**Comment by yvele** [comment]
@bengrunfeld Yep it's way more easy.. jest replaces mocha, chai, istanbul, nyc, rewire.... and is way more powerfull 😉  (

**Comment by alexgalin** [comment]
any update here?
